web/source/osk/visualKeyboard.ts

```diff
@@ -172,53 +183,46 @@ namespace com.keyman.osk {
     }
 
     public postInsert(): void { }
-
+ 
+    /**
+     * The configured width for this VisualKeyboard.  May be `undefined` or `null`
+     * to allow automatic width scaling. 
+     */
     get width(): number {
-      if(this._width) {
-        return this._width;
-      } else {
-        this.loadSizeFromCookie();
-        return this.width;
-      }
+      return this._width;
     }
 
+    /**
+     * The configured height for this VisualKeyboard.  May be `undefined` or `null`
+     * to allow automatic height scaling. 
+     */
     get height(): number {
-      if(this._height) {
-        return this._height;
-      } else {
-        this.loadSizeFromCookie();
-        return this.height;
-      }
+      return this._height;
     }
 
-    protected loadSizeFromCookie() {
```

Comment:
No more cookies for `VisualKeyboard`; `OSKManager` now fully manages the OSK's width and height, passing updates along to `VisualKeyboard` as needed.

`refreshLayout` is used once those values are available.  Always after `_Load` (particularly, 'after the OSK is inserted into the DOM').

Reply:
If no more cookies, then what remembers the keyboard size and position between page loads on e.g. keymanweb.com? (not saying that cookies within kmw are the right solution, but we need to make sure that there is some pathway for migration)

Reply:
> No more cookies **_for `VisualKeyboard`_** [...]

I never said no more cookies for `OSKManager` - that's where those references belong and remain.